streaming/base/dataloader.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2022 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Streaming DataLoader."""
+
+from typing import Any, Dict, Iterator, Optional
+
+from torch import Tensor
+from torch.utils.data import DataLoader
+from transformers.tokenization_utils_base import BatchEncoding
+
+from streaming.base.dataset import Dataset
+from streaming.base.world import World
+
+
+class StreamingDataLoader(DataLoader):
```

Comment:
When would user instantiate the `StreamingDataLoader` class? is it when user is using vanilla pytorch model (without composer) and wants to do checkpointing/resumption ??

Reply:
yes